azure-pipelines.yml

```diff
@@ -407,142 +407,131 @@ jobs:
-  - bash: |
-      pip install --force-reinstall wheel
-  - task: Bash@3
-    displayName: 'Run Test Automation'
-    inputs:
-      targetType: 'filePath'
-      filePath: ./scripts/ci/test_automation.sh
-    env:
-      REDUCE_SDK: 'True'
-      AZURE_CLI_TEST_TARGET_PROFILE: '2019-03-01'
-      ADO_PULL_REQUEST_TARGET_BRANCH: $(System.PullRequest.TargetBranch)
+        python -m venv env
+        chmod +x env/bin/activate
+        source env/bin/activate
 
-- job: RunAutomation20180301
-  displayName: Run Automation, Profile 2018-03-01
-  dependsOn: BuildPythonWheel
+        pip install -q azdev
+        azdev --version
 
-  pool:
-    vmImage: 'ubuntu-16.04'
-  steps:
-  - task: UsePythonVersion@0
-    displayName: 'Use Python 3.6'
-    inputs:
-      versionSpec: 3.6
-  - task: DownloadPipelineArtifact@1
-    displayName: 'Download Python Wheels'
-    inputs:
-      TargetPath: '$(Build.ArtifactStagingDirectory)/pypi'
-      artifactName: pypi
-  - bash: |
-      pip install --force-reinstall wheel
-  - task: Bash@3
-    displayName: 'Run Test Automation'
-    inputs:
-      targetType: 'filePath'
-      filePath: ./scripts/ci/test_automation.sh
-    env:
-      AZURE_CLI_TEST_TARGET_PROFILE: '2018-03-01'
-      ADO_PULL_REQUEST_TARGET_BRANCH: $(System.PullRequest.TargetBranch)
+        azdev setup -c ./
+      displayName: Setup Env
+    - bash: |
+        set -ev
 
-- job: RunAutomationPython3_8
-  displayName: Run Automation Python 3.8
+        source env/bin/activate
+
+        if [[ "$(System.PullRequest.TargetBranch)" != "" ]]; then
+          azdev test --series --repo=./ --src=HEAD --tgt=origin/$(System.PullRequest.TargetBranch) --cli-ci
+        else
+          azdev test --series
+        fi
+      displayName: "Automation Test"
+      env:
+        ADO_PULL_REQUEST_LATEST_COMMIT: $(System.PullRequest.TargetBranch)
+
+- job: AutomationTest20190301
+  displayName: Automation Test (Profile 2019-03-01)
   dependsOn: BuildPythonWheel
-  timeoutInMinutes: 90
+  timeoutInMinutes: 30
 
   pool:
     vmImage: 'ubuntu-16.04'
+  strategy:
+    matrix:
+      Python36:
+        python.version: '3.6'
+      Python38:
+        python.version: '3.8'
   steps:
-  - task: UsePythonVersion@0
-    displayName: 'Use Python 3.8'
-    inputs:
-      versionSpec: 3.8
-  - task: DownloadPipelineArtifact@1
-    displayName: 'Download Python Wheels'
-    inputs:
-      TargetPath: '$(Build.ArtifactStagingDirectory)/pypi'
-      artifactName: pypi
-  - bash: |
-      pip install --force-reinstall wheel
-  - task: Bash@3
-    displayName: 'Run Test Automation'
-    inputs:
-      targetType: 'filePath'
-      filePath: ./scripts/ci/test_automation.sh
-    env:
-      ADO_PULL_REQUEST_TARGET_BRANCH: $(System.PullRequest.TargetBranch)
+    - task: UsePythonVersion@0
+      inputs:
+        versionSpec: '$(python.version)'
+      displayName: 'Use Python $(python.version)'
+    - bash: |
+        set -ev
+
+        git fetch origin dev
```

Comment:
why use a git fetch here?

Reply:
Removed since git checkout in ADO will fetch all target branches.